src/hooks/useIsValidRecoveryExecution.ts

```diff
@@ -0,0 +1,192 @@
+      }
+    })()
+  }, [isRecoverer, safeTx, wallet, web3ReadOnly, safe.chainId, delayModifierAddress])
+}
+
+// Hook to check the validity of an existing recovery proposal before it is executed
+// e.g. a Account setup that was previously submitted to Delay Modifier
+export function useIsValidRecoveryExecuteNextTx(recovery: RecoveryQueueItem): AsyncResult<boolean> {
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const web3ReadOnly = useWeb3ReadOnly()
+  const { isExecutable } = useRecoveryTxState(recovery)
+
+  return useAsync(() => {
+    if (!isExecutable || !wallet?.address || !web3ReadOnly) {
+      return
+    }
+
+    return (async () => {
+      const provider = getPatchedSignerProvider(wallet, safe.chainId, web3ReadOnly)
+      const delayModifier = getModuleInstance(KnownContracts.DELAY, recovery.address, provider)
+
+      const signer = provider.getSigner()
+      const contract = delayModifier.connect(signer)
+
+      try {
+        const { to, value, data, operation } = recovery.args
+
+        await contract.callStatic.executeNextTx(to, value, data, operation)
+
+        return true
+      } catch (error) {
+        throw error
+      }
+    })()
+  }, [isExecutable, recovery.address, recovery.args, safe.chainId, wallet, web3ReadOnly])
+}
+
+// Hook to check the validity of skipping expired recovery proposals before it is executed
+// e.g. Account proposals that were previously submitted to Delay Modifier but have since expired
+export function useIsValidRecoverySkipExpired(recovery: RecoveryQueueItem): AsyncResult<boolean> {
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const web3ReadOnly = useWeb3ReadOnly()
+  const { isExpired } = useRecoveryTxState(recovery)
+
+  return useAsync(() => {
+    if (!isExpired || !wallet?.address || !web3ReadOnly) {
+      return
+    }
+
+    return (async () => {
+      const provider = getPatchedSignerProvider(wallet, safe.chainId, web3ReadOnly)
+      const delayModifier = getModuleInstance(KnownContracts.DELAY, recovery.address, provider)
+
+      const signer = provider.getSigner()
+      const contract = delayModifier.connect(signer)
+
+      try {
+        await contract.callStatic.skipExpired()
+
+        return true
+      } catch (error) {
+        throw error
+      }
+    })()
+  }, [isExpired, recovery.address, safe.chainId, wallet, web3ReadOnly])
+}
+
+// Hook to check the validity of what happens to a Safe after a recovery proposal is executed
+// e.g. Account setup that was previously submitted to Delay Modifier and will be executed
+export function useIsValidRecoveryExecution(recovery: RecoveryQueueItem): AsyncResult<boolean> {
+  const wallet = useWallet()
+  const { safe } = useSafeInfo()
+  const web3ReadOnly = useWeb3ReadOnly()
+  const { isExecutable } = useRecoveryTxState(recovery)
+
+  return useAsync(() => {
+    if (!isExecutable || !wallet?.address || !web3ReadOnly) {
+      return
+    }
+
+    return (async () => {
```

Comment:
Why do we need this self executing block? `useAsync` should already catch possible errors and return them.